manifests/init.pp

```diff
@@ -12,155 +12,65 @@
-  validate_string($pam_krb5_ccache_type)
-  validate_string($pam_silent)
-  validate_string($pam_mkhomedir)
-  validate_string($smb_workgroup)
-  validate_string($smb_realm)
-  validate_string($smb_encrypt_passwords)
-  validate_string($smb_server_string)
-  validate_string($smb_security)
-  validate_string($smb_log_file)
-  validate_string($smb_printcap_name)
-  validate_string($smb_printing)
-  validate_string($smb_winbind_enum_users)
-  validate_string($smb_winbind_enum_groups)
-  validate_string($smb_winbind_nss_info)
-  validate_string($smb_winbind_normalize_names)
-  validate_string($smb_winbind_separator)
-  validate_string($smb_template_homedir)
-  validate_string($smb_template_shell)
-  validate_string($smb_idmap_config_default_backend)
-  validate_string($krb5_default)
-  validate_string($krb5_kdc)
-  validate_string($krb5_admin_server)
-  validate_string($krb5_ticket_lifetime)
-  validate_string($krb5_renew_lifetime)
-  validate_string($oddjobd_homdir_mask)
+    contain ::winbind::install
+    contain ::winbind::config
+    contain ::winbind::service
 
-  # arrays
-  validate_array($pam_require_membership_of)
-  validate_array($smb_includes_files)
-
-  # numbers
-  if ( !is_numeric($pam_warn_pwd_expire) ) {
-    fail('pam_warn_pwd_expire must be a number')
-  }
-
-  if ( !is_numeric($smb_log_level) ) {
-    fail('smb_log_level must be a number')
-  }
-
-  if ( !is_numeric($smb_syslog) ) {
-    fail('smb_syslog must be a number')
-  }
-
-  if ( !is_numeric($smb_max_log_size) ) {
-    fail('smb_max_log_size must be a number')
-  }
-
-  if ( !is_numeric($smb_idmap_config_default_range_start) ) {
-    fail('smb_idmap_config_default_range_start must be a number')
-  }
-
-  if ( !is_numeric($smb_idmap_config_default_range_end) ) {
-    fail('smb_idmap_config_default_range_end must be a number')
-  }
-
-  if ( !is_numeric($smb_idmap_config_default_rangesize) ) {
-    fail('smb_idmap_config_default_rangesize must be a number')
-  }
-
-
-  # booleans
-  if ( !is_bool($smb_winbind_use_default_domain) ) {
-    fail('smb_winbind_use_default_domain must be a true or false')
-  }
-
-  if ( !is_bool($smb_winbind_offline_logon) ) {
-    fail('smb_winbind_offline_logon must be a true or false')
-  }
-
-  if ( !is_bool($krb5_dns_lookup_realm) ) {
-    fail('krb5_dns_lookup_realm must be a true or false')
-  }
-
-  if ( !is_bool($krb5_dns_lookup_kdc) ) {
-    fail('krb5_dns_lookup_kdc must be a true or false')
-  }
-
-  if ( !is_bool($krb5_forwardable) ) {
-    fail('krb5_forwardable must be a true or false')
-  }
-  validate_bool($manage_oddjob_service)
-  validate_bool($manage_messagebus_service)
-
-
-  # use the install -> config -> service model
-  anchor {'::winbind::start': }
-  -> class { '::winbind::install': }
-  -> class { '::winbind::config': }
-  -> class { '::winbind::service': }
-  -> anchor {'::winbind::end': }
+  Class['::winbind::install']
+  -> Class['::winbind::config']
+  ~> Class['::winbind::service']
```

Comment:
I think I just happen to miss that small detail about what the `~>` is for though its clearly in the [docs](https://docs.puppet.com/puppet/5.0/lang_relationships.html#syntax-chaining-arrows). I believe it can just be changed to a hyphen, since resource relationships are already being handled in config.